.github/workflows/build.yml

```diff
@@ -28,6 +28,10 @@ jobs:
           # rust dependencies
           export CARGO_HOME="$HOME/.cargo"
           curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain 1.75.0
+          # nim dependencies
+          export CHOOSENIM_CHOOSE_VERSION=2.0.4
+          curl https://nim-lang.org/choosenim/init.sh -sSf | sh -s -- -y
+          export PATH=$HOME/.nimble/bin:$PATH
```

Comment:
if we change the image to ubuntu-23.10 , we can install nim from apt.  Looks like it is version 1.16.14.  So maybe downloading from nim-lang is a good idea.  the curl-scripted kind of installation is a supply chain attack vector though.  Same could be said for the installation of rust right above it 🤔

Reply:
The problem is in the final-assembly step, because I checked this step is working fine and installing nim correctly.
I see steps like these in final-assembly 
```
      - name: Download bls12-381
        uses: actions/download-artifact@v3
        with:
          name: bls12-381 native build artifacts
          path: bls12-381/build/
```

I checked building with nim 2.0.4 which worked fine for me.
In constantine lib - `Constantine can be compiled by Nim v1.6.x, v2.0.2 and v2.0.4 but not Nim v2.0.0 (due to a compile-time evaluation crash)` 
But installing with apt won't change much as even choose-nim is working fine

Reply:
Yup, I would expect it will work fine, but giving the contents of a URL the privilege to run anything in the shell is a really sketchy way to install dependencies.  We are doing the same thing for rust, and we really shouldn't.

Reply:
1.6.x (latest is 1.6.20) works fine for Constantine and it's tested in CI. No benefits performance-wise for latest Nim versions. Constantine uses low-level Nim and has no library dependencies and even no std-lib dependencies or allocator dependencies to be extremely hard to supply-chain attack (i.e. introducing a compiler bug).
Hence the features that might justify upgrading a compiler are more related to its compile-time capabilities and type system which Constantine uses extensively.